GetIntoTeachingApiTests/Contracts/TeacherTrainingAdviserTests.cs

```diff
@@ -59,7 +61,21 @@ public async void Contract(string scenario)
 
             await WriteInitialOutputFile(outputFile, requestJson);
 
-            requestJson.Should().BeEquivalentTo(File.ReadAllText(outputFile));
+            var request = SortEntities(JArray.Parse(requestJson));
+            var snapshot = SortEntities(JArray.Parse(File.ReadAllText(outputFile)));
+
+            request.Should().HaveCount(snapshot.Count());
+            request.Should().BeEquivalentTo(snapshot);
+        }
+
+        private static JArray SortEntities(JArray entities)
+        {
+            for(var i = 0; i < entities.Count; i++)
```

Comment:
Wondering if this for loop is left over from something else you tried and could be a foreach?

Reply:
Ah yeah that could be cleaned up a bit, I tried a whole bunch of things to make it recursively sort `JArray` and `JObject` but it made it way less readable overall and we don't seem to nest more than one level anyway. I'll fixup a commit.